### PR TITLE
Feat/generated client resolves references

### LINKED ISSUE #3010: Resolve references in generated queries
## Summary

As a Tina Dev I want to have my references resolved in the generated queries  so that I don't need to use GraphQL for this.

### Expected Results

- Can add a CLI frag that can pass in the max generated depth (with a default of 3)
- Resolve references in generated queries ([code here](https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/graphql/src/builder/index.ts#L541))

### Notes

Time box to a day.

### PR DESCRIPTION
**Pull request by logan-anderson**
fixes #3010

### CODE REVIEW

File: .changeset/three-owls-melt.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    config: {
+        client: {
+            referenceDepth: 3
+        },
+    }
+    // ...
+})
+```
+
+To get the old behaver set referenceDepth to `1`.
```
Comment (kldavis4) [inline .changeset/three-owls-melt.md:20]:
```suggestion
To get the old behavior set referenceDepth to `1`.
```

**Review by kldavis4 — COMMENTED** [review]
@logan-anderson I noticed that the generated files are smaller (ie __generated__/_schema.json) - is that to be expected? (I was thinking it would be larger)

**Comment by kldavis4** [comment]
@logan-anderson what's the best way to validate this?

File: experimental-examples/tina-cloud-starter/.tina/schema.ts [hunk 2]
```diff
@@ -66,13 +66,10 @@ const BlocksTemplate: TinaTemplate = {
 
 const schema = defineSchema({
   config: {
+    client: {
+      referenceDepth: 5,
+    },
     media: {
-      // If you wanted cloudinary do this
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply experimental-examples/tina-cloud-starter/.tina/schema.ts:70]: Nope

File: packages/@tinacms/graphql/src/builder/index.ts [hunk 3]
```diff
@@ -54,6 +57,7 @@ export const createBuilder = async ({
  * storing a reference to how we can resolve that type when we come across it.
  */
 export class Builder {
+  private _MAXDEPTH: number
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply packages/@tinacms/graphql/src/builder/index.ts]: I made the name better 😄. There was no reason for it to be upper case with an underscore.

**Comment by logan-anderson** [comment]
> @logan-anderson I noticed that the generated files are smaller (ie **generated**/_schema.json) - is that to be expected? (I was thinking it would be larger)

@kldavis4. Ill take a look into this. They should not be smaller. The _schema.json should not really change (unless you add the config). Frags.gql, and types.ts should be larger.

File: experimental-examples/tina-cloud-starter/.tina/schema.ts [hunk 4]
```diff
@@ -87,13 +90,9 @@ const schema = defineSchema({
       format: "mdx",
       fields: [
         {
-          label: "3 layer nesting",
```
[the remark this answers is not in the document]
Comment (logan-anderson) [reply experimental-examples/tina-cloud-starter/.tina/schema.ts]: I update the schema.

**Comment by kldavis4** [comment]
> > @logan-anderson what's the best way to validate this?
> 
> @kldavis4 Use the generated client localy. (it is set up in the tina-cloud-starter) and make a request to a collection with references and see if the data that comes back resolves them.

@logan-anderson Does the TinaCMS admin use the generated client? If not I am not sure the best way to see the requests that are being made in getStaticProps/getStaticPaths without adding some logging.

**Review by kldavis4 — APPROVED** [review]

**Pull request merged**